agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -743,7 +743,13 @@ impl DeriveSchema for Expression {
             Expression::Literal(ref l) => derive_schema_for_literal(l),
             Expression::Ref(Ref::FieldRef(f)) => {
                 let path = f.split(".").map(|s| s.to_string()).collect::<Vec<String>>();
-                let schema = get_schema_for_path_mut(&mut state.result_set_schema, path);
+                // If the user has rebound the CURRENT variable, we should use that schema instead of the result set schema to find any
+                // path.
```

Comment:
```suggestion
                // If the user has rebound the CURRENT variable, we should use that schema instead of the result set schema to find any
                // path. The reason we do this is because a field reference `$<field>` is equivalent to `$$CURRENT.<field>.
```
First of all, wow, great catch here 🔥 .

Feel free to change the wording. I think we should explain why this is relevant since it is not immediately obvious. Most people maintaining this codebase (well, most people in general), probably won't realize that all field references are implicitly `$$CURRENT`-prefixed variables references!

Reply:
Sadly I can't just commit suggestions since they won't be signed. Really annoying.

Reply:
Can you still push your own commit with an updated comment here?